### PR TITLE
author name and package name snake case lower case only. ipfshash dat…

### PR DESCRIPTION
**Pull request by solarw**
author name and package name snake case lower case only. ipfshash datatype test fix

## Proposed changes

author name and package name snake case lower case only. ipfshash datatype test fix

## Fixes

some tests

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

DELETE INCLUSIVE THIS AND BELOW FOR STANDARD PR
------

## Release summary

Version number: [e.g. 1.0.1]

## Release details

Describe in short the main changes with the new release.

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side), from `develop`
- [ ] Lint and unit tests pass locally and in CI
- [ ] I have checked the fingerprint hashes are correct by running (`aea packages lock --check`)
- [ ] I have regenerated the latest API docs
- [ ] I built the documentation and updated it with the latest changes
- [ ] I have added an item in `HISTORY.md` for this release
- [ ] I bumped the version number in the `aea/__version__.py` file.
- [ ] I bumped the version number in every Docker image of the repo and published it. Also, I built and published them with tag `latest`  
      (check the READMEs of [`aea-develop`](../develop-image/README.md#publish) 
      and [`aea-user`](../user-image/README.md#publish))
- [ ] I have pushed the latest packages to the registry.
- [ ] I have uploaded the latest `aea` to PyPI.
- [ ] I have uploaded the latest plugins to PyPI.

## Further comments

Write here any other comment about the release, if any.

### CODE REVIEW

File: aea/helpers/base.py [hunk 1]
```diff
@@ -249,51 +249,66 @@ class SimpleId(RegexConstrainedString):
...
+    ValueError: Value an_Id does not match the regular expression re.compile('[a-z_][a-z0-9_]{0,127}')
     """
 
     REGEX = re.compile(SIMPLE_ID_REGEX)
 
 
 class IPFSHash(RegexConstrainedString):
     """
-    A simple identifier.
+    A IPFSHash identifier.
```
Comment (solarw) [inline aea/helpers/base.py:280]: fix for tests for IPFSHash

File: aea/cli/utils/package_utils.py [hunk 2]
```diff
@@ -133,13 +139,17 @@ def _is_valid_author_handle(author: str) -> bool:
...
+    ...
+    False
+    >>> _is_valid_author_handle("and_not_This_one")
+    ...
+    False
 
     :param author: author name
     :return: bool indicating whether author name is valid
     """
-    if re.fullmatch(PublicId.AUTHOR_REGEX, author) is None:
```
Comment (solarw) [inline aea/cli/utils/package_utils.py:140]: dont like check internals exposed and used here.
this code should not know about regex and how to use them for checking

File: aea/cli/utils/package_utils.py [hunk 3]
```diff
@@ -117,9 +117,15 @@ def validate_package_name(package_name: str) -> None:
...
     click.exceptions.BadParameter: this-is-not is not a valid package name.
 
+    >>> validate_package_name("This_is_not")
+    Traceback (most recent call last):
+    ...
+    click.exceptions.BadParameter: This_is_not is not a valid package name.
+
     :param package_name: the package name
     """
-    if re.fullmatch(PublicId.PACKAGE_NAME_REGEX, package_name) is None:
```
Comment (solarw) [inline aea/cli/utils/package_utils.py:122]: this code should not know about regex and how to use them for checking

**Comment by solarw** [comment]
@DavidMinarsch @angrybayblade 
this change can be potentially quite dangerous.
some packages do not meet new REGEX will fail to work (think about 3rd parts)

**Review by DavidMinarsch — COMMENTED** [review]
Please update the check copyright script in the same way as on open-autonomy

**Comment by DavidMinarsch** [comment]
> @DavidMinarsch @angrybayblade this change can be potentially quite dangerous. some packages do not meet new REGEX will fail to work (think about 3rd parts)

Yes it is very much breaking. Although I prefer doing this change at this nascent stage. Later it will be impossible to do.

**Review by angrybayblade — APPROVED** [review]

File: docs/p2p-connection.md [hunk 4]
```diff
@@ -58,7 +58,11 @@ Create a second AEA:
 ``` bash
 aea create my_other_aea
 cd my_other_aea
+<<<<<<< HEAD
+aea add connection valory/p2p_libp2p:0.1.0:bafybeif5evkmahtr2ioqi5ndjxns3w4thy2ihgsol4skatiafmn66345t4 --remote
+=======
```
[the remark this answers is not in the document]
Comment (solarw) [reply docs/p2p-connection.md]: fixed

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**